### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by rahul1995**
## Purpose

To provide documentation for the connector, including its usage and setup.

Issue Title: Introduce a Ballerina connector for Dropbox REST API
Issue link: https://github.com/ballerina-platform/ballerina-library/issues/7054

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

**Comment by rahul1995** [comment]
@lnash94 @shafreenAnfar @NipunaRanasinghe please review.

File: README.md [hunk 1]
```diff
@@ -8,15 +8,81 @@
...
+
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+Navigate to https://www.dropbox.com/developers/apps and select “Create app”.
```
Comment (lnash94) [inline README.md:22]: Shall we elaborate on the steps more, by giving UI examples?
you can find the reference here : https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/README.md

File: README.md [hunk 2]
```diff
@@ -8,15 +8,81 @@
...
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+Navigate to https://www.dropbox.com/developers/apps and select “Create app”. 
+1. Choose an API  
+2. Choose the type of access you need 
+3. Give your app a name  
+4. Choose the Dropbox account that will own your app
```
Comment (lnash94) [inline README.md]: Shall we elaborate on the steps more, by giving UI examples?
you can find the reference here: https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/README.md

Comment (rahul1995) [reply README.md]: @lnash94 done!

**Comment by rahul1995** [comment]
@lnash94 made the changes, please review!

File: ballerina/Package.md [hunk 3]
```diff
@@ -1,14 +1,83 @@
...
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+1. Navigate to https://www.dropbox.com/developers/apps and sign up to create an account (or log in if you already have an account)
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-developer-portal.png alt="Dropbox Developer Portal" style="width: 70%;">
+
+2. select “Create app”.
```
Comment (lnash94) [inline ballerina/Package.md]:
```suggestion
2. Select “Create app”.
```

File: ballerina/Module.md [hunk 4]
```diff
@@ -1,14 +1,83 @@
...
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+1. Navigate to https://www.dropbox.com/developers/apps and sign up to create an account (or log in if you already have an account)
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-developer-portal.png alt="Dropbox Developer Portal" style="width: 70%;">
+
+2. select “Create app”.
```
Comment (lnash94) [inline ballerina/Module.md]:
```suggestion
2. Select “Create app”.
```

File: ballerina/Package.md [hunk 5]
```diff
@@ -1,14 +1,83 @@
...
+```
+
+### Step 4: Run the Ballerina application
+
+```ballerina
+bal run
+```
+
 
 ## Examples
```
Comment (lnash94) [inline ballerina/Package.md:82]: Shall we update this section with the references you have written as examples?

Comment (rahul1995) [reply ballerina/Package.md:82]: I have taken care of examples related documentation in the[ Examples PR](https://github.com/ballerina-platform/module-ballerinax-dropbox/pull/6).

File: README.md [hunk 6]
```diff
@@ -8,15 +8,84 @@
...
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+1. Navigate to https://www.dropbox.com/developers/apps and sign up to create an account (or log in if you already have an account)
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-developer-portal.png alt="Dropbox Developer Portal" style="width: 70%;">
+
+2. select “Create app”.
```
Comment (lnash94) [inline README.md]:
```suggestion
2. Select “Create app”.
```

File: README.md [hunk 7]
```diff
@@ -8,15 +8,84 @@
...
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">
+3. Choose an API, type of access you need, and give your app a name.
```
Comment (lnash94) [inline README.md]:
```suggestion
<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">

3. Choose an API, type of access you need, and give your app a name.
```

File: ballerina/Module.md [hunk 8]
```diff
@@ -1,14 +1,83 @@
...
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+1. Navigate to https://www.dropbox.com/developers/apps and sign up to create an account (or log in if you already have an account)
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-developer-portal.png alt="Dropbox Developer Portal" style="width: 70%;">
+
+2. Select “Create app”.
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">
+3. Choose an API, type of access you need, and give your app a name.
```
Comment (lnash94) [inline ballerina/Module.md]:
```suggestion
2. Select “Create app”.
<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">

3. Choose an API, type of access you need, and give your app a name.
```

File: ballerina/Package.md [hunk 9]
```diff
@@ -1,14 +1,83 @@
...
+To use the Dropbox connector, you must have access to the Dropbox API. Dropbox uses OAuth 2.0, an open specification, to authorize access to data. To get an OAuth token from Dropbox to enable Dropbox Connector to access your Dropbox account via the API you’ll need to create a new app on the DBX Platform.
+
+### Creating an App on the DBX Platform
+
+1. Navigate to https://www.dropbox.com/developers/apps and sign up to create an account (or log in if you already have an account)
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-developer-portal.png alt="Dropbox Developer Portal" style="width: 70%;">
+
+2. Select “Create app”.
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">
+3. Choose an API, type of access you need, and give your app a name.
```
Comment (lnash94) [inline ballerina/Package.md]:
```suggestion
2. Select “Create app”.
<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-dropbox/main/docs/setup/resources/dropbox-apps-dashboard.png alt="Dropbox Apps Dashboard" style="width: 70%;">

3. Choose an API, type of access you need, and give your app a name.
```

**Review by lnash94 — DISMISSED** [review]

File: README.md [hunk 10]
```diff
@@ -8,15 +8,85 @@
...
+token = "<Enter your token here>"
+```
+
+2. Create a dropbox:ConnectionConfig with the obtained access token and initialize the connector with it.
+
+```ballerina
+configurable http:BearerTokenConfig & readonly authConfig = ?
+dropbox:ConnectionConfig config = {auth: authConfig};
+
+final dropbox:Client dropbox = check new(config, serviceUrl = "https://api.dropboxapi.com/2");
```
Comment (NipunaRanasinghe) [inline README.md]: We can mention this approach instead, which looks cleaner
```suggestion
configurable string token = ?;
dropbox:ConnectionConfig config = {auth: {token}};

final dropbox:Client dropbox = check new(config, serviceUrl = "https://api.dropboxapi.com/2");
```

**Review by lnash94 — DISMISSED** [review]

File: README.md (hunk 10 above)
Comment (NipunaRanasinghe) [reply README.md]: @rahul1995 need to add the same changes to the `Module.md` and `Package.md` files

Comment (rahul1995) [reply README.md]: @NipunaRanasinghe fixed.

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]

**Pull request merged**